internal/overlord/logstate/syslog.go

```diff
@@ -0,0 +1,184 @@
+package logstate
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/canonical/pebble/internal/plan"
+)
+
+func init() {
+	RegisterLogBackend("syslog", func(t *plan.LogTarget) (LogBackend, error) { return NewSyslogBackend(t.Location) })
+}
+
+func validateSyslog(t *plan.LogTarget) error {
+	b, err := NewSyslogBackend(t.Location)
+	if b != nil {
+		b.Close()
+	}
+	return err
+}
+
+// SyslogBackend takes writes and formats them according to RFC5424.  The formatted syslog messages
+// are then forwarded to the underlying syslog server.  SyslogWriter is safe for concurrent writes
+// and use.
+type SyslogBackend struct {
+	version        int
+	host           string
+	msgid          string
+	priority       int
+	pid            string
+	structuredData string // TODO: use this for future log labels
+
+	// mu helps manage access to network connection-related internal state.  This should be
+	// read-locked whenever using the syslog server connection and write-locked whenever
+	// replacing/reconnecting/etc the syslog server connection.
+	mu            sync.RWMutex
+	conn          net.Conn
+	address       *url.URL
```

Comment:
Same comment above as address. It's a connection url and the naming can be better to not imply that it is always an address.

Reply:
So `connURL` or something?